modules/openxr/scene/openxr_composition_layer.cpp

```diff
@@ -756,13 +763,27 @@ void OpenXRCompositionLayer::_validate_property(PropertyInfo &p_property) const
 	}
 }
 
+XROrigin3D *OpenXRCompositionLayer::_get_xrorigin3d_ancestor() const {
+	Node *parent = get_parent();
+	while (parent != nullptr) {
+		XROrigin3D *origin = Object::cast_to<XROrigin3D>(parent);
+		if (origin != nullptr) {
+			return origin;
+		}
+
+		parent = parent->get_parent();
+	}
+
+	return nullptr;
+}
+
 PackedStringArray OpenXRCompositionLayer::get_configuration_warnings() const {
 	PackedStringArray warnings = Node3D::get_configuration_warnings();
 
-	if (is_visible() && is_inside_tree()) {
-		XROrigin3D *origin = Object::cast_to<XROrigin3D>(get_parent());
+	if (is_visible_in_tree() && is_inside_tree()) {
+		XROrigin3D *origin = _get_xrorigin3d_ancestor();
 		if (origin == nullptr) {
-			warnings.push_back(RTR("OpenXR composition layers must have an XROrigin3D node as their parent."));
+			warnings.push_back(RTR("OpenXR composition layers must have an XROrigin3D node as their ancestor."));
```

Comment:
Wait, I thought the idea was that you could put it anywhere? Why does it still need to have `XROrigin3D` as an ancestor?